packages/common/src/hardhat/tasks/identifiers.js

```diff
@@ -24,3 +24,68 @@ task("whitelist-identifiers", "Whitelist identifiers from JSON file")
 
     await _whitelistIdentifier(web3, id, identifierWhitelist, deployer);
   });
+
+task("migrate-identifiers", "Adds all whitelisted identifiers on one IdentifierWhitelist to another")
+  .addParam("from", "The contract from which to query a whitelist of identifiers.", "", types.string)
+  .addOptionalParam("to", "The contract on which to whitelist new identifiers.", "", types.string)
+  .setAction(async function (taskArguments, hre) {
+    const { deployments, getNamedAccounts, web3 } = hre;
+    const { deployer } = await getNamedAccounts();
+    const { from, to } = taskArguments;
+
+    const IdentifierWhitelist = await deployments.get("IdentifierWhitelist");
+    const oldWhitelist = new web3.eth.Contract(IdentifierWhitelist.abi, from);
+
+    const identifiersToWhitelist = [];
+    const addedIdentifierEvents = await oldWhitelist.getPastEvents("SupportedIdentifierAdded", { fromBlock: 0 });
+
+    // Filter out identifiers that are not currently whitelisted.
+    let isIdentifierSupportedPromises = [];
+    addedIdentifierEvents.forEach((_event) => {
+      isIdentifierSupportedPromises.push(
+        oldWhitelist.methods.isIdentifierSupported(_event.returnValues.identifier).call()
+      );
+    });
+    (await Promise.all(isIdentifierSupportedPromises)).forEach((supported, i) => {
+      if (supported) {
+        identifiersToWhitelist.push(addedIdentifierEvents[i].returnValues.identifier);
+      }
+    });
+
+    // Create table with results to display to user:
+    let resultsTable = identifiersToWhitelist.map((id) => {
+      return {
+        identifierToWhitelist: id,
+        utf8: web3.utils.hexToUtf8(id),
+      };
+    });
+
+    if (to) {
+      const newWhitelist = new web3.eth.Contract(IdentifierWhitelist.abi, to);
+      isIdentifierSupportedPromises = [];
+      identifiersToWhitelist.forEach((id) => {
+        isIdentifierSupportedPromises.push(newWhitelist.methods.isIdentifierSupported(id).call());
+      });
+
+      const isIdentifierSupportedResults = await Promise.all(isIdentifierSupportedPromises);
```

Comment:
nit: I think this can be simplified
```suggestion
      const isIdentifierSupportedResults = await Promise.all(identifiersToWhitelist.map(id => newWhitelist.methods.isIdentifierSupported(id).call()));
```

Reply:
Done nice refactor